packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -77,10 +77,14 @@ export class PeerExchangeDiscovery
     );
   };
 
-  constructor(components: Libp2pComponents, options: Options = {}) {
+  constructor(
+    components: Libp2pComponents,
+    pubsubTopics: PubsubTopic[],
+    options: Options = {}
+  ) {
     super();
     this.components = components;
-    this.peerExchange = new WakuPeerExchange(components);
+    this.peerExchange = new WakuPeerExchange(components, pubsubTopics);
```

Comment:
would it be better to have `pubsubTopics` required within `BaseProtocol` so that they don't need to be passed for every protocol instantiation?